pycqed/measurement/pulse_sequences/multi_qubit_tek_seq_elts.py

```diff
@@ -2152,3 +2152,375 @@ def ro_dynamic_phase_seq(hard_sweep_dict, qbp_name, qbr_names,
+    swept_pulses_with_prep = \
+        [add_preparation_pulses(p, operation_dict, qubit_names, **prep_params)
+         for p in swept_pulses]
+    seq = pulse_list_list_seq(swept_pulses_with_prep, seq_name, upload=False)
+
+    # add calibration segments
+    seq.extend(cal_points.create_segments(operation_dict, **prep_params))
+
+    # reuse sequencer memory by repeating readout pattern
+    [seq.repeat_ro(f"RO {qbn}", operation_dict) for qbn in qubit_names]
+
+    log.debug(seq)
+    if upload:
+        ps.Pulsar.get_instance().program_awgs(seq)
+
+    return seq, np.arange(seq.n_acq_elements())
+
+
+def n_qubit_qscale_seq(qubit_names, operation_dict, sweep_points, cal_points,
+                       upload=True, for_ef=False,
+                       last_ge_pulse=False, prep_params=dict()):
+    '''
+    DRAG pulse calibration sequence for n qubits.
+    Args:
+        qubit_names:     list of qubit names
+        operation_dict:  operation_dict for all qubit in qubit_names
+        sweep_points:    instance of SweepPoints class
+        cal_points:      instance of CalibrationPoints class
+        upload:          whether to upload sequence to instrument or not
+        n:               number of pulses (1 is conventional Rabi)
+        for_ef:          whether to do rabi between ef transition
+        last_ge_pulse:   whether to use a ge pulse at the end of each segment
+            for a rabi between ef transition
+        prep_params:     qubit preparation params
+    Returns:
+        sequence (Sequence): sequence object
+        segment_indices (list): array of range of n_segments including
+            calibration_segments. To be used as sweep_points for the MC.
+    '''
+
+    seq_name = 'n_qubit_qscale_sequence'
+
+    # Operations
+    qscale_base_ops = [['X90', 'X180'], ['X90', 'Y180'], ['X90', 'mY180']]
+
+    # add DRAG calibration segments
+    final_pulses = []
+    for base_ops in qscale_base_ops:
+        pulse_list = []
+        for qbn in qubit_names:
+            qscale_ops = add_suffix(base_ops, "_ef" if for_ef else "")
+            if for_ef:
+                qscale_ops = ['X180'] + qscale_ops
+                if last_ge_pulse:
+                    qscale_ops += ["X180"]
+            qscale_ops = add_suffix(qscale_ops, " " + qbn)
+
+            qscale_pulses = [deepcopy(operation_dict[op]) for op in qscale_ops]
+            qscale_pulses[0]['ref_pulse'] = 'segment_start'
+            # name and reference swept pulse
+            for i in range(len(base_ops)):
+                idx = (1 if for_ef else 0) + i
+                qscale_pulses[idx]["name"] = f"Qscale_{i}_{qbn}"
+            pulse_list += qscale_pulses
+        pulse_list += generate_mux_ro_pulse_list(qubit_names, operation_dict)
+
+        params_to_sweep = {
+            f'Qscale_*_{qbn}.motzoi': list(sweep_points[0].values())[j][0]
+            for j, qbn in enumerate(qubit_names)}
+        # print(params_to_sweep)
+        # print(len(pulse_list))
```

Comment:
delete print statements (applies to others also which are commented)

Reply:
thanks!